libp2p/Host.cpp

```diff
@@ -127,16 +134,18 @@ void Host::stop()
     // such tasks may involve socket reads from Capabilities that maintain references
     // to resources we're about to free.
 
-    // ignore if already stopped/stopping, at the same time,
-    // signal run() to prepare for shutdown and reset m_timer
+    // ignore if already stopped/stopping, at the same time
+    // indicates that the network is shutting down
     if (!m_run.exchange(false))
         return;
 
-    {
-        unique_lock<mutex> l(x_runTimer);
-        while (m_timer)
-            m_timerReset.wait(l);
-    }
+    // stopping io service allows running manual network operations for shutdown
+    // and also stops blocking worker thread, allowing worker thread to exit
+    m_ioService.stop();
+
+    // Close the node table socket and cancel deadline timers. This effectively stops
+    // discovery, even during subsequent io service polling
+    nodeTable()->stop();
```

Comment:
I don't think I need to do this - we can let the node table continue to run since it's not deleted until the Host instance is deleted, meaning there's no risk of a UAF. @gumb0 : Thoughts?

Reply:
Yeah, that's what I suggested, because I think it might be a good idea to tell it to not run the handlers' code even if they are still called